### DISCUSSION TITLE
SDLC processes and software architectures used in research projects (Session two - Room two)

### DISCUSSION THREAD

**Discussion by junranlei** [RSEAA/RSEAA.github.io#139] [Discussion topics for #RSEAA24]
Discussion can include (please feel free to make other suggestions too):
SDLC templates and workflows;
DevOps processes, platforms and toolsets;
Software architectures used in research projects (monolithic, microservice, centralised, decentralised, SOLID, client-server, Jamstack...?)

**Comment by madhurabl** [comment]
For those familiar with Drupal and headless technologies, could you provide insights on the best approaches? Specifically, we're looking at a setup where the frontend handles only forms for data capture and interactive reporting. What technologies or strategies would you recommend for interactive reporting?

**Reply by junranlei** [reply]
> For those familiar with Drupal and headless technologies, could you provide insights on the best approaches? Specifically, we're looking at a setup where the frontend handles only forms for data capture and interactive reporting. What technologies or strategies would you recommend for interactive reporting?

I am not sure about interactive reporting but you can check out Next.js for Drupal https://next-drupal.org/ and template https://github.com/wunderio/next-drupal-starterkit, or other packages for Headless CMS here: https://jamstack.org/generators/, or just use React directly, check out the comparison here too: https://manmash.consulting/blog/headless-drupal/
